OSDP - Don't use TEST_RANDOM_GENERATOR

ceolin opened the pull request:
- Do not use TEST_RANDOM_GENERATOR
- Do not select unecessarily ENTROPY_GENERATOR
- Do not select wrong SERIAL options

Commit:
osdp: Do not use TEST_RANDOM_GENERATOR

TEST_RANDOM_GENERATOR should not be used in subsystems because it does
not generate truly random numbers. This is indented to be used only
for tests purpose.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
osdp: Do not select ENTROPY_GENERATOR

Entropy generator is not used in this subsystem so there is no need to
select it.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

nashif commented on samples/subsys/mgmt/osdp/sample.yaml:
```diff
@@ -0,0 +1,12 @@
+      - frdm_k64f
+
+    # FIXME: exclude due to build error
+    platform_exclude: qemu_xtensa
```
look at this

https://github.com/zephyrproject-rtos/zephyr/pull/27913#issuecomment-685153646

ceolin replied:
cool, just remove that commit.

jenmwms replied:
Thanks! I'll focus on osdp/sample.yaml there (#27913). For Kconfig, the comment^^ mentioned to also remove `SERIAL_HAS_DRIVER` and `UART_INTERRUPT_DRIVEN`. Do we need that too?

ceolin replied:
I haven't checked that to be honest. Will look

Commit:
osdp: Remove invald serial option

SERIAL_HAS_DRIVER and UART_INTERRUPT_DRIVEN should be selected by the
driver. If something needs is should dependson these options.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>
